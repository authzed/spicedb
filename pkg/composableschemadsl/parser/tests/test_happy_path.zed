test DocumentCaveatedTest {
  relationships {
    post:one reader user:beatrice
    post:one caveated_reader user:beatrice with likes_harry_potter
    post:one temporary_reader user:claire with temporary_reader {"selected_day": "tuesday"}
  }
  assertions {
    post:one    view_as_fan    user:beatrice {"likes": true}  // Positive assertion
    post:one    !view_as_fan   user:beatrice {"likes": false} // Negative assertion
    post:one    ?view_as_fan   user:beatrice                  // Conditional assertion

    post:one    view_temporarily    user:claire {"current_day": "tuesday"}
    post:one    !view_temporarily   user:claire {"current_day": "wednesday"}
  }
  expected {
    post:one view {
      user:beatrice is reader of post:one
    }
    post:one view_as_fan {
      user:beatrice[...] is caveated_reader of post:one
    }
  }
}
