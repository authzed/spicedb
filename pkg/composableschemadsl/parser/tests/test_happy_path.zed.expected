NodeTypeFile
  end-rune = 26
  input-source = test syntax happy path test
  start-rune = 0
  child-node =>
    NodeType(21)
      end-rune = 26
      input-source = test syntax happy path test
      start-rune = 0
      test-name = DocumentCaveatedTest
      child-node =>
        NodeTypeError
          end-rune = 26
          error-message = Expected end of statement or definition, found: TokenTypeKeyword
          error-source = relationships
          input-source = test syntax happy path test
          start-rune = 30
    NodeTypeError
      end-rune = 26
      error-message = Unexpected token at root level: TokenTypeKeyword
      error-source = relationships
      input-source = test syntax happy path test
      start-rune = 30